README.md

```diff
@@ -1,123 +1,323 @@
+    * [S3 Scanner](#s3-scanner)
+
+* [How to adapt this pipeline for your use cases](#how-to-adapt-this-pipeline-for-your-use-cases)
+
+* [Troubleshooting](#troubleshooting)
+
+* [Advanced Topics](#advanced-topics)
+
+* [Disclaimer](#disclaimer)
+
+
+## Architecture
+The solution comprises two types of pipelines (based on the DLP Transformation type). Refer [Dataflow DAG](#dataflow-dag) for pipeline job graph:
+1. Inspect and Deidentify
+2. ReIdentify
+
+### Inspection and De-identification
+![Reference Architecture](diagrams/inspect-deid-architecture.png)
+
+The pipeline can be used for CSV, TSV, Avro, JSONL files stored in either GCS or AWS S3 bucket. It uses State and Timer API for efficient batching to process the files in optimal manner.
+The results of inspection/the deidentified data is written to BigQuery table. 
+
+### Re-identification
+![Reference Architecture](diagrams/reid-architecture.png)
+
+The pipeline for reidentification workflow is used to read data from BigQuery table and publish the re-identified data in a secure pub sub topic.
+## Concepts
+
+1. [Cloud Data Loss Prevention - Quick Start & Guides](https://cloud.google.com/dlp/docs/dlp-bigquery)
+2. [De-identification and re-identification of PII in large-scale datasets using Cloud DLP](https://cloud.google.com/solutions/de-identification-re-identification-pii-using-cloud-dlp).
+3. [Create & Manage Cloud DLP Configurations](https://cloud.google.com/dlp/docs/creating-job-triggers).
+4. [Automated Dataflow Pipeline to De-identify PII Dataset](https://cloud.google.com/dataflow/docs/guides/templates/provided/dlp-text-to-bigquery).
+5. [Validate Dataset in BigQuery and Re-identify using Dataflow](https://cloud.google.com/solutions/validating-de-identified-data-bigquery-re-identifying-pii-data).
+6. [Inspecting storage and databases for sensitive data](https://cloud.google.com/dlp/docs/inspecting-storage)
+7. [Dataflow Pipeline Options](https://cloud.google.com/dataflow/docs/reference/pipeline-options)
+8. [Cloud DLP Quotas and Limits](https://cloud.google.com/dlp/limits)
+
+
+## Tutorial
+
+### Pre-requisites
+
+1. Create a new project on Google Cloud Platform.
+
+2. Use the link below to open Google Cloud Shell.
+
+   [![Open in Cloud Shell](http://gstatic.com/cloudssh/images/open-btn.svg)](https://console.cloud.google.com/cloudshell/editor?cloudshell_git_repo=https://github.com/GoogleCloudPlatform/dlp-dataflow-deidentification.git)
+
+3. Run the following commands to set up the data tokenization solution in your GCP project. 
 
 ```
 gcloud config set project <project_id>
-sh deploy-data-tokeninzation-solution.sh
+sh deploy-data-tokeninzation-solution-v2.sh
 ```
 
- You can run some quick [validations](https://cloud.google.com/solutions/validating-de-identified-data-bigquery-re-identifying-pii-data#validating_the_de-identified_dataset_in_bigquery) in BigQuery table to check on tokenized data.  
+Script (deploy-data-tokenization-solution-v2.sh) handles following topics:
+
```

Comment:
The script also performs the below steps as well and I think we should be mentioning them - 
Gathers & exports project ID.
Creates key ring called demo-key-ring for KMS wrapped key
Enables DLP service API, CloudKMS, BQ, GCS, Dataflow
Creates a service account(demo-service-account) & adds roles & permissions mentioned in dlp_tokenizing_runner_permissions.yaml
Creates default inspection, DeId & Reid templates and uploads them to GCS inside staging folder & also copies the same locally
Emits set_env.sh as well as exports all the environment variables used in the following below commands.

Also we should call out that enabling the extra services might incur relevant charges and provide the link to pricing calculator here

Reply:
Most of these steps are mentioned already. Giving more details would be unnecessary as the purpose here is to explain a high level (end goals) of the script. Steps like enabling DLP API are pre-requisites for some of the end goals.

Reply:
Yes, but customer should know what additional services are being enabled, resources utilized and billed. I'll let @annadata share his thoughts

Reply:
done